From 90071037eaf841ae016d8a3c437a9cc1e178c1ff Mon Sep 17 00:00:00 2001
Message-Id: <90071037eaf841ae016d8a3c437a9cc1e178c1ff.1430045673.git.chang-joon.lee@intel.com>
From: Deepak S <deepak.s@intel.com>
Date: Thu, 23 Apr 2015 22:15:26 +0530
Subject: [PATCH] REVERTME [VPG]: drm/i915: Fix gfx pm suspend/resume sequence

As part of S3/S0ix sequence, "ALLOW_WAKE_BIT" cleared to avoid
force/demand wake request after the suspend. But with current Gfx PM sequence,
Even after clearing the "ALLOW_WAKE_BIT" we allow register access :(
Spec says "ALLOW_WAKE_BIT" should be cleared at the end of the
sequence. This patch matches the sequence as per the spec.

Revert Condition: This is a portion fix from Upstream. Next forklift
should have proper implementation of switcheroo/legacy suspend paths

Issue: GMINL-8548
Change-Id: Ib5ca40971d9eefb82a71d422badbfd801f932a9c
Signed-off-by: Deepak S <deepak.s@intel.com>
---
 drivers/gpu/drm/i915/i915_drv.c |    5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_drv.c b/drivers/gpu/drm/i915/i915_drv.c
index 7b1f97a..d3d0ae1 100644
--- a/drivers/gpu/drm/i915/i915_drv.c
+++ b/drivers/gpu/drm/i915/i915_drv.c
@@ -628,13 +628,12 @@ static int i915_drm_freeze(struct drm_device *dev)
 
 	dev_priv->suspend_count++;
 
-	ret = intel_suspend_complete(dev_priv);
+	intel_display_set_init_power(dev_priv, false);
 
+	ret = intel_suspend_complete(dev_priv);
 	if (ret)
 		WARN(1, "Suspend complete failed: %d\n", ret);
 
-	intel_display_set_init_power(dev_priv, false);
-
 	kobject_uevent_env(&dev->primary->kdev->kobj, KOBJ_CHANGE, envp);
 
 	return 0;
-- 
1.7.9.5

